src/Lykke.AlgoStore.Api/Controllers/AlgoController.cs

```diff
@@ -59,5 +66,190 @@ public AlgoController(IAlgoStoreClientDataService clientDataService)
+
+            data.ClientId = User.GetClientId();
+
+            var result = await _clientDataService.SaveAlgoRatingAsync(data);
+
+            var response = Mapper.Map<AlgoRatingModel>(result);
+
+            return Ok(response);
+        }
+
+        [HttpGet("userAlgoRating")]
+        [SwaggerOperation("userAlgoRating")]
+        [ProducesResponseType(typeof(AlgoRatingModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(BaseErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetUserAlgoRating(string algoId, string clientId)
+        {
+            if (string.IsNullOrWhiteSpace(clientId))
+                clientId = User.GetClientId();
+
+            var result = await _clientDataService.GetAlgoRatingForClientAsync(algoId, clientId);
+
+            if (result == null)
+                return NotFound();
+
+            var response = Mapper.Map<AlgoRatingModel>(result);
+
+            return Ok(response);
+        }
+
+        [HttpPost("addToPublic")]
+        [SwaggerOperation("addToPublic")]
+        [ProducesResponseType(typeof(PublicAlgoDataModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(BaseErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> AddToPublic([FromBody] PublicAlgoDataModel model)
+        {
+            var clientId = User.GetClientId();
+            var data = Mapper.Map<PublicAlgoData>(model);
+
+            var result = await _clientDataService.AddToPublicAsync(data, clientId);
+
+            var response = Mapper.Map<PublicAlgoDataModel>(result);
+
+            return Ok(response);
+
+        }
+
+        [HttpPost("removeFromPublic")]
+        [SwaggerOperation("removeFromPublic")]
+        [ProducesResponseType(typeof(PublicAlgoDataModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(BaseErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> RemoveFromPublic([FromBody] PublicAlgoDataModel model)
+        {
+            var clientId = User.GetClientId();
+            var data = Mapper.Map<PublicAlgoData>(model);
+
+            var result = await _clientDataService.RemoveFromPublicAsync(data, clientId);
+
+            var response = Mapper.Map<PublicAlgoDataModel>(result);
+
+            return Ok(response);
+
+        }
+
+        [HttpGet("algoMetadata")]
+        [SwaggerOperation("GetAlgoMetadata")]
+        [ProducesResponseType(typeof(AlgoDataInformationModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(BaseErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
```

Comment:
should be renamed, no need for Metadata work anymore

Reply:
Can be named GetAlgoInformation, both swagger name and method name should be renamed